modules/openapi-generator/src/test/resources/3_0/oneoOfDiscriminator.yaml

```diff
@@ -0,0 +1,201 @@
+openapi: 3.0.1
+info:
+   title: fruity
+   version: 0.0.1
+paths:
+   /:
+      get:
+         responses:
+            '200':
+               description: desc
+               content:
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/FruitAllOfDisc'
+components:
+   schemas:
+    FruitType:
+      properties:
+        fruitType:
+          type: string
+      required:
+      - fruitType
+    FruitInlineDisc:
+      oneOf:
+        - type: object
+          required:
+            - seeds
+            - fruitType
+          properties:
+            seeds:
+              type: integer
+            fruitType:
+              type: string
+        - type: object
+          required:
+            - length
+            - fruitType
+          properties:
+            length:
+              type: integer
+            fruitType:
+              type: string
+      discriminator:
+        propertyName: fruitType
+    FruitInlineInlineDisc:
+      oneOf:
+        - type: object
+          required:
+            - seeds
+          properties:
+            seeds:
+              type: integer
+          oneOf:
+            - type: object
+              properties:
+                fruitType:
+                  type: string
+              required:
+              - fruitType
+        - type: object
+          required:
+            - length
+          properties:
+            length:
+              type: integer
+          oneOf:
+            - type: object
+              properties:
+                fruitType:
+                  type: string
+              required:
+              - fruitType
+      discriminator:
+        propertyName: fruitType
+    FruitReqDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleReqDisc'
+        - $ref: '#/components/schemas/BananaReqDisc'
+      discriminator:
```

Comment:
This should imply common inheritance between Apple and Banana.

Reply:
I don't understand what you mean here. Can you be more specific?

Reply:
Yeah, you're right. I misread it as an allOf.
Just keeping you on your toes 🤪